### PR TITLE
feat: inline comment tag

### PR DESCRIPTION
**Pull request by jg-rp**
This pull request implements the inline comment tag, as found in [Ruby Liquid version 5.4.0](https://github.com/Shopify/liquid/pull/1498). At the time of writing, Ruby Liquid version 5.4.0 is "unreleased", but inline comments [are already available in Shopify store templates](https://github.com/Shopify/liquid/pull/1498#issuecomment-1143594442).

Inline comments look like this:
```
{% # This is a comment %}
{%-
  # Comments can span multiple lines,
  # but every line must start with a hash.
-%}
```

And inside `liquid` tags:
```
{% liquid
  # This is a comment
  echo "Hello"
%}
```
You might also be interested in [this modified Prism language definition for Liquid](https://github.com/jg-rp/liquid/blob/docs/src/theme/prism-liquid-custom.js).

### CODE REVIEW

File: src/parser/tokenizer.ts [hunk 1]
```diff
@@ -234,6 +233,17 @@ export class Tokenizer {
     return new IdentifierToken(this.input, begin, this.p, this.file)
   }
 
+  readTagName (): string {
+    this.skipBlank()
+    const begin = this.p
+    // Handle inline comment tags
+    if (this.input[this.p] === '#') {
+      return this.input.slice(begin, ++this.p)
+    }
```
Comment (harttle) [inline src/parser/tokenizer.ts:242]: can we call `readIdentifier` here?

**Review by harttle — APPROVED** [review]

**Comment by harttle** [comment]
We'll need to add [docs for this tag](https://github.com/harttle/liquidjs/tree/master/docs/source/tags) and [website sidebar](https://github.com/harttle/liquidjs/blob/4a0186d62a0c9d16afe75d26861e575cc5f51228/docs/source/_data/sidebar.yml#L79). I can help update the zh-cn translation.

File: src/parser/tokenizer.ts (hunk 1 above)
Comment (jg-rp) [reply src/parser/tokenizer.ts:242]: Can do, after the check for `#`. '#' being an invalid character for identifiers.

I was expecting a minor performance improvement by avoiding the creation of an `IdentifierToken` for every tag name, as `readIdentifier` does. Both `TagToken` and `LiquidTagToken` were immediately calling `getText()` on the `IdentifierToken` to get the string. I've not benchmarked it though.

**Comment by jg-rp** [comment]
> We'll need to add [docs for this tag](https://github.com/harttle/liquidjs/tree/master/docs/source/tags) and [website sidebar](https://github.com/harttle/liquidjs/blob/4a0186d62a0c9d16afe75d26861e575cc5f51228/docs/source/_data/sidebar.yml#L79). I can help update the zh-cn translation.

I'll have a go at docs for this tag within the next day or two.

File: src/parser/tokenizer.ts (hunk 1 above)
Comment (jg-rp) [reply src/parser/tokenizer.ts]: Benchmarking showed no discernible performance improvement. Refactored accordingly in 200fb36.

Comment (harttle) [reply src/parser/tokenizer.ts]: Makes sense, I think it’s OK either way.

**Pull request merged**